docs/source/api.yaml

```diff
@@ -1626,6 +1653,80 @@ paths:
           description: Returns a confirmation of the logout
           schema:
             $ref: '#/definitions/logout'
+  /api/applications:
+    get:
+      summary: List
+      operationId: findApplications
+      tags:
+        - Applications
+      description: |
+        Get a list of application objects for all applctations the user is
+        authorized to see.
+      responses:
+        '200':
+          description: Returns metadata for all applications the user is authorized to see
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/apiClient'
+    post:
+      summary: Create
+      operationId: createApplication
+      tags:
+        - Applications
+      description: |
+        Creates a new application.
+      parameters:
+        - name: body
+          in: body
+          description: Application object to create
+          required: true
+          schema:
+            $ref: '#/definitions/apiClient'
+      responses:
+        '201':
+          description: Returns the newly created Application
+          schema:
+            $ref: '#/definitions/apiClient'
+  '/api/applications/{applicationId}':
+    parameters:
+      - $ref: '#/parameters/applicationId'
+    get:
+      summary: Get
+      description: |
+        Get the details about the given application.
+      operationId: findApplicationById
```

Comment:
The rest endpoint i set up was findApplicationByName(). From a user standpoint that makes more sense since they will likely know the name but the id is autogenerated.

Reply:
Are we sure that's how we want to handle it? That breaks from the id convention of all our other rest endpoints and requires application names across all users to be unique.

Reply:
Application names are intended to be unique and there is a constraint in the JPA set up that requires the name to be unique.

That being said, if we're using id in other places then maybe it makes sense here. But the user will need a way to retrieve the id based on name.

Reply:
Actually, now that I've said that I have a question about how this feature is presented in the UI. If the findByApplicationId is meant to be used by selecting from a list of applications the logged in user has access to then I guess it doesn't matter.

Reply:
I have changed that in a new branch. We have the ability to find by name but other methods use an ID